crates/neon/src/sys/tag.rs

```diff
@@ -137,3 +139,37 @@ pub unsafe fn check_object_type_tag(env: Env, object: Local, tag: &super::TypeTa
 pub unsafe fn is_bigint(env: Env, val: Local) -> bool {
     is_type(env, val, napi::ValueType::BigInt)
 }
+
+pub unsafe fn is_map(env: Env, val: Local) -> bool {
```

Comment:
In my opinion, there isn't a great reason to put this in `sys`. A lot of the `sys` patterns like `sys::tag` go back to when Neon supported a C++ backend (prior to Node-API).

The current conventions are to put this code alongside the safe implementation. This also makes it easier to use _other_ safe wrappers instead of falling back to unsafe. Less unsafe code is needed. Unfortunately, the `ValueInternal` trait uses pointers, so _some_ unsafe is needed, but only for getting a `Context` and `Handle`.

Reply:
I opened https://github.com/neon-bindings/neon/pull/1081 to allow implementing `ValueInternal::is_typeof` without any unsafe.

Reply:
So you mean I should entirely remove `sys::tag::is_map` and instead put it content directly inside `impl private::ValueInternal for JsMap`'s `is_typeof` ?

Reply:
Yep, exactly!